File: src/adapters/pi/defaults.ts
import type { AdapterConfig } from '../../core/types.js';

/** pi adapter 的唯一标识（写进 CategorySnapshot.adapterId / AdapterSnapshot.adapterId）。 */
export const PI_ADAPTER_ID = 'pi';

/**
 * pi adapter 默认配置（计划 §1.5 逐字落地）。
 *
 * - root 用 '~' 前缀，由 paths 层负责展开（本模块不碰 HOME）。
 * - 7 个分类：settings / skills / extensions / agents / models / prompts / themes。
 * - ignore 是 adapter 级忽略（相对 root 的路径 glob）。
 */
export const DEFAULT_PI_ADAPTER: AdapterConfig = {
  root: '~/.pi/agent',
  enabled: true,
  categories: {
    settings: { paths: ['settings.json', 'keybindings.json'], mode: 'merge' },
    skills: { paths: ['skills/'], mode: 'mirror' },
    extensions: { paths: ['extensions/'], mode: 'mirror', exclude: ['*cache*'] },
    agents: { paths: ['agents/'], mode: 'mirror' },
    models: { paths: ['models.json'], mode: 'merge', excludeKeys: ['apiKeys'] },
    prompts: { paths: ['prompts/'], mode: 'mirror' },
    themes: { paths: ['themes/'], mode: 'mirror' },
  },
  ignore: [
    'auth.json',
    'trust.json',
    'sessions/',
    'npm/',
    'git/',
    'tmp/',
    'bin/',
    '*.bak',
    '*.bak-*',
    '*.bak*',
    '*.log',
    'run-history.jsonl',
  ],
};

File: internal/adapter/pi/defaults.go
package pi

import "github.com/zzjcool/homer-cli/internal/core"

// PIAdapterID is the stable adapter identifier written to snapshots and
// homer.json.
const PIAdapterID = "pi"

// PI_ADAPTER_ID is kept as a source-compatible spelling for callers ported
// directly from the archived TypeScript adapter.
const PI_ADAPTER_ID = PIAdapterID

func boolPtr(value bool) *bool { return &value }

// DefaultPIAdapter is the frozen pi configuration from the legacy TypeScript
// source (the sole source of truth for this adapter's categories and ignore
// list).
var DefaultPIAdapter = core.AdapterConfig{
	Root:    "~/.pi/agent",
	Enabled: boolPtr(true),
	Categories: map[string]core.CategoryConfig{
		"settings": {
			Paths: []string{"settings.json", "keybindings.json"},
			Mode:  core.SyncMode("merge"),
		},
		"skills": {
			Paths: []string{"skills/"},
			Mode:  core.SyncMode("mirror"),
		},
		"extensions": {
			Paths:   []string{"extensions/"},
			Mode:    core.SyncMode("mirror"),
			Exclude: []string{"*cache*"},
		},
		"agents": {
			Paths: []string{"agents/"},
			Mode:  core.SyncMode("mirror"),
		},
		"models": {
			Paths:       []string{"models.json"},
			Mode:        core.SyncMode("merge"),
			ExcludeKeys: []string{"apiKeys"},
		},
		"prompts": {
			Paths: []string{"prompts/"},
			Mode:  core.SyncMode("mirror"),
		},
		"themes": {
			Paths: []string{"themes/"},
			Mode:  core.SyncMode("mirror"),
		},
	},
	Ignore: []string{
		"auth.json",
		"trust.json",
		"sessions/",
		"npm/",
		"git/",
		"tmp/",
		"bin/",
		"*.bak",
		"*.bak-*",
		"*.bak*",
		"*.log",
		"run-history.jsonl",
	},
}
